src/entity/object/EnderCrystal.php

```diff
@@ -80,20 +96,33 @@ protected function initEntity(CompoundTag $nbt) : void{
 		$this->setHealth(1);
 
 		$this->setShowBase($nbt->getByte(self::TAG_SHOWBASE, 0) === 1);
+
+		$beamX = $nbt->getInt(self::TAG_BLOCKTARGET_X, 0);
+		$beamY = $nbt->getInt(self::TAG_BLOCKTARGET_Y, 0);
+		$beamZ = $nbt->getInt(self::TAG_BLOCKTARGET_Z, 0);
+
+		if($beamX !== 0 || $beamY !== 0 || $beamZ !== 0){
```

Comment:
This is probably not a good idea. The beam could actually target 0,0,0.

Reply:
When beam position 0, 0, 0 is sent to client, the client doesn't draw beam

Reply:
That's not the point. This should be checked by checking the presence of the tags. Do not fill in defaults.